storage/interceptable_repository.go

```diff
@@ -350,11 +355,14 @@ func (ir *queryScopedInterceptableRepository) Update(ctx context.Context, obj ty
 
 func (ir *queryScopedInterceptableRepository) UpdateLabels(ctx context.Context, objectType types.ObjectType, objectID string, labelChanges types.LabelChanges, criteria ...query.Criterion) error {
 	byID := query.ByField(query.EqualsOperator, "id", objectID)
-	obj, err := ir.repositoryInTransaction.Get(ctx, objectType, byID)
+	result, err := ir.repositoryInTransaction.ListNoLabels(ctx, objectType, byID)
 	if err != nil {
 		return err
 	}
-
+	if result == nil || result.Len() == 0 {
+		return util.ErrNotFoundInStorage
+	}
+	obj := result.ItemAt(0)
```

Comment:
maybe also check that you have a single result and fail if you got multiple? i know you are querying by ID, but someone might mess the query itself